examples/user_interfaces/embedding_in_tk_sgskip.py

```diff
@@ -30,18 +31,35 @@
 toolbar = NavigationToolbar2Tk(canvas, root, pack_toolbar=False)
 toolbar.update()
 
-
 canvas.mpl_connect(
     "key_press_event", lambda event: print(f"you pressed {event.key}"))
 canvas.mpl_connect("key_press_event", key_press_handler)
 
-button = tkinter.Button(master=root, text="Quit", command=root.quit)
+button_quit = tkinter.Button(master=root, text="Quit", command=root.quit)
+
+
+def update_frequency():
+    # random frequency
+    f = randint(1, 5)
+
+    # retrieve Axes object and update plots there
+    ax = fig.axes[0]
+    ax.clear()
+    ax.plot(t, 2 * np.sin(2 * np.pi * f * t))
+
+    # required to update canvas and attached toolbar!
+    canvas.draw()
+
+
+button_update = tkinter.Button(
```

Comment:
Thanks, this seems useful!  Any chance you could make this a slider instead?  If not, can you change the button name to "Change Freq" or something more indicative of what it does?

Reply:
Done, looks like this now:

![image](https://user-images.githubusercontent.com/12762439/118258371-0c632500-b4b0-11eb-8970-d4b6d60d05b6.png)

Reply:
Cool.  I think if you added @QuLogic suggestion below that would be very nice as well.  What you are doing is the simplest thing to do, but what he is suggesting is "best-practice" when dealing with animations.